### PR TITLE
Change title for plugins documentation

### LINKED ISSUE #6729: doc page title is confusing
The page that explains all sorts of plugins (Scheduler Plugins, Worker Plugin, even Nanny Plugins!) is titled "Scheduler Plugins". This is confusing.

It affects what's listed in the docs sidebar:

image

It also affects search results:

image

For a while I thought "why aren't there any docs about worker plugins?!" but eventually I looked at the "scheduler plugin" page and saw the information I wanted.

### PR DESCRIPTION
**Pull request by scharlottej13**
Closes #6729 

- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

In addition to changing the title, I also added a few sentences at the top for a summary, and a video snippet that seemed like a relevant example.

cc @jrbourbeau

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: docs/source/plugins.rst [hunk 1]
```diff
@@ -1,3 +1,12 @@
+Plugins
+~~~~~~~
+
+Plugins enable you to run custom Python code for certain events. You can use plugins
```
Comment (jrbourbeau) [inline docs/source/plugins.rst:4]:
```suggestion
Dask's plugin system enables you to run custom Python code for certain events. You can use plugins
```

File: docs/source/plugins.rst [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+that are specific to the scheduler, the workers, or the nanny. A worker plugin, for example,
```
Comment (jrbourbeau) [inline docs/source/plugins.rst:5]:
```suggestion
that are specific to schedulers, workers, or nannies. A worker plugin, for example,
```

File: docs/source/plugins.rst [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
+allows you to run custom Python code on all your workers at any event in the worker's lifecycle.
```
Comment (jrbourbeau) [inline docs/source/plugins.rst:6]:
```suggestion
allows you to run custom Python code on all your workers at certain event in the worker's lifecycle (e.g. when the worker process is started).
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @scharlottej13! Overall this looks great -- just left a few minor comments

**Review by jacobtomlinson — APPROVED** [review]
This looks great! I agree with @jrbourbeau's tweaks so I hope you don't mind but I've committed them.

**Comment by jrbourbeau** [comment]
Thanks @scharlottej13 @jacobtomlinson

**Comment by scharlottej13** [comment]
Thanks @jrbourbeau @jacobtomlinson !

**Pull request merged**